### ISSUE TITLE
Collision of event-entity in item merge event

### ISSUE THREAD

**Issue by TheLimeGlass** [SkriptLang/Skript#4787]
Skript 2.6.2
PaperSpigot 1.18.2

### Bug Description

Successfully parses:
```vb
on item merge:
	broadcast "%type of event-entity%"
	broadcast "%event-item%"
	message "Testing, will this send?" to player
	message "Testing, will this send?" to event-entity
```
The message effects are never sent to player. The actual item entity in the event is the entity receiving the messages.

Very strange that specifying `player` doesn't make Skript check all the entities involved in the event for a player type, but that's probably relating to another issue. Also adding the fact that there is no player in the event...

### Expected Behavior

The message effects being sent to the player should be sent to the player and not the item entity.

Proposed change would be doing a similar fix as https://github.com/SkriptLang/Skript/pull/4540 to where when using event-entity in the item merge event, you get greeted with an error message stating to use the event value expression of the merged item. Something like `merged item` as an expression syntax, and an error message like `Use 'merged item' and/or 'player' in the item merge event.`

Although player would need a custom event value expression too since `player` is currently erroring. I'll see if using the built in parameter of the event value fixes using `player` or if I should make a dedicated event value expression for that or fix Skript being able to find `player` out of all the entities.

### Errors or Screenshots

<img width="288" alt="Untitled" src="https://user-images.githubusercontent.com/16087552/172368052-0e448152-1310-4726-bb80-598d79c84dd3.png">
Output from the broadcast effects, this is all that is printed.

**Comment by TheLimeGlass** [comment]
Waiting on merge of https://github.com/SkriptLang/Skript/pull/4858
